Lib/glyphsLib/builder.py

```diff
@@ -340,6 +344,10 @@ def set_custom_params(ufo, parsed=None, data=None, misc_keys=(), non_info=()):
         if name.startswith('openTypeOS2Win') and value < 0:
             value = -value
 
+        # Some Glyphs files store weightClass/widthClass as string values.
+        if name in ('openTypeOS2WeightClass', 'openTypeOS2WidthClass'):
+            value = int(value)
```

Comment:
Just tried to set `weightClass` custom parameter to a float in Glyphs.app, and the latter didn't complain, the `weightClass` is stored like this in the .glyphs file:
```
{
name = weightClass;
value = "111.9";
}
```

After exporting the font, the OS/2 table has `<usWeightClass value="111"/>`.

So it seems that Glyphs.app is simply truncating the value, as in `int(value)`, and not `round(value)`.